src/minerva_lib/crop.py

```diff
@@ -0,0 +1,298 @@
+from functools import reduce
+import numpy as np
+from .blend import composite_channel
+from . import skimage_inline as ski
+
+
+def scale_image_nearest_neighbor(source, factor):
+    ''' Resizes an image by a given factor using nearest neighbor pixels.
+
+    Arguments:
+        source: A 2D or 3D numpy array to resize
+        factor: The ratio of `out` shape over `source` shape
+
+    Returns:
+        A numpy array with the resized `source`
+    '''
+
+    if factor <= 0:
+        raise ValueError('Scale factor must be above zero')
+
+    factors = [factor, factor, 1]
+
+    out_shape = [int(round(a * b)) for a, b in zip(source.shape, factors)]
+    out_max_x = out_shape[1] - 1
+    out_max_y = out_shape[0] - 1
+    out = np.zeros(out_shape)
+
+    for source_x in range(0, source.shape[1]):
+        for source_y in range(0, source.shape[0]):
+            out_x = min(int(round(source_x * factor)), out_max_x)
+            out_y = min(int(round(source_y * factor)), out_max_y)
+            out[out_y, out_x] = source[source_y, source_x]
+
+    return out
+
+
+def get_optimum_pyramid_level(input_shape, level_count,
+                              max_size=None, round_up=True):
+    ''' Gives a pyramid level that scales a region near a maximum size.
```

Comment:
I don't understand this description. I think a better summary would be something like.

"Calculates the optimum pyramid level to use rendering an image region into an output image with a given bound".